### PR TITLE
AzureTestPlanTask: Implemented support for Go Language

### PR DESCRIPTION
**Pull request by t-shreysingh**
**Task name**: AzureTestPlanTask: Implemented support for Go Language

**Description**: Successfully installing gotestsum package and running test cases from different files. https://dev.azure.com/t-shreysingh/Go%20project/_build/results?buildId=240&view=logs&j=275f1d19-1bd8-5591-b06b-07d489ea915a
Resolved the conversation of the older PR, which is closed because of merge conflicts https://github.com/microsoft/azure-pipelines-tasks/pull/19980
**Documentation changes required:** (Y/N) <Please mark if documentation changes are required>

**Added unit tests:** (Y/N) <Please mark if unit tests were added or updated according changes>

**Attached related issue:** (Y/N) <Please add link to related issue here>

**Checklist**:
- [ ] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [ ] Checked that applied changes work as expected
image
image
image
image

### CODE REVIEW

File: Tasks/AzureTestPlanV0/Invokers/goinvoker.ts [hunk 1]
```diff
@@ -1,15 +1,35 @@
...
+            const GoTestPath = utils.separateGoPath(tests);
+            const GoTestName = utils.separateGoTestName(tests);
+            const argument = `--junitfile TEST-Go${i}-junit.xml -- ${GoTestPath} -v -run ^${GoTestName}$`;
+            const status = await executeGoCommand(goPath, argument);
+            if (status != 0) {
+                finalStatus = 1;
+            }
+            tl.debug("Tests Run Successfully");
+        } catch (error) {
+            console.error(`Error executing test case:`, error);
```
Comment (triptijain2112) [inline Tasks/AzureTestPlanV0/Invokers/goinvoker.ts:29]: console.error(`Error executing test case:`, error);

add test case name as well

File: Tasks/AzureTestPlanV0/Invokers/goinvoker.ts [hunk 2]
```diff
@@ -1,15 +1,35 @@
...
+    for (let tests of testsToBeExecuted) {
+        try {
+            const GoTestPath = utils.separateGoPath(tests);
+            const GoTestName = utils.separateGoTestName(tests);
+            const argument = `--junitfile TEST-Go${i}-junit.xml -- ${GoTestPath} -v -run ^${GoTestName}$`;
+            const status = await executeGoCommand(goPath, argument);
+            if (status != 0) {
+                finalStatus = 1;
+            }
+            tl.debug("Tests Run Successfully");
```
[the remark this answers is not in the document]
Comment (t-shreysingh) [reply Tasks/AzureTestPlanV0/Invokers/goinvoker.ts]: updated

File: Tasks/AzureTestPlanV0/Invokers/goinvoker.ts (hunk 1 above)
Comment (t-shreysingh) [reply Tasks/AzureTestPlanV0/Invokers/goinvoker.ts]: updated

**Review by triptijain2112 — APPROVED** [review]

**Review by adityashahms — APPROVED** [review]
Looks good to me

**Comment by adityashahms** [comment]
I hope apart from data driven tests, linking of test point and test result is populating as expected in test plan UI?

**Pull request merged**